examples/stable_diffusion_v2/inference/libs/sd_infer.py

```diff
@@ -0,0 +1,83 @@
+from time import time
+from tqdm import tqdm
+import mindspore as ms
+from .modules import MaskLatent
+
+class SD_Text2Img:
+    def __init__(self, text_encoder, unet, scheduler, vae_decoder, guidance_scale=7.5, num_inference_steps=50):
+        super(SD_Text2Img, self).__init__()
+        self.text_encoder = text_encoder
+        self.unet = unet
+        self.scheduler = scheduler
+        self.vae_decoder = vae_decoder
+        self.do_classifier_free_guidance = guidance_scale > 1.0
+        self.num_inference_steps = ms.Tensor(num_inference_steps, ms.int32)
+
+    def __call__(self, prompts, negative_prompts, start_sample, timesteps):
+        prompt_embeds = self.text_encoder(prompts)
+        negative_prompt_embeds = self.text_encoder(negative_prompts)
+        latents = start_sample
+        iterator = tqdm(timesteps, desc="Stable Diffusion Sampling", total=len(timesteps))
+        for i, t in enumerate(iterator):
+            ts = ms.Tensor(t, ms.int32)
+            noise_pred = self.unet(latents, ts, prompt_embeds, negative_prompt_embeds)
+            latents = self.scheduler(noise_pred, ts, latents, self.num_inference_steps)
+        image = self.vae_decoder(latents)
+        return image
+
+
+class SD_Img2Img:
```

Comment:
There are many repetitions among these classes. Can we combine them? For example, `SD_Text2Img` and `SD_Img2Img` differ in the latent space encoding only. I think we can simply create a method for extracting latent space and override it inside `SD_Img2Img` class:

```python
def _get_latents():
    return class_specific_latent_space
```